src/nbl/video/COpenGLCommandBuffer.cpp

```diff
@@ -754,19 +748,21 @@ namespace video
             case impl::ECT_SET_EVENT:
             {
                 auto& c = cmd.get<impl::ECT_SET_EVENT>();
-                // TODO
+                //https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/vkCmdSetEvent2KHR.html
+                // A memory dependency is defined between the event signal operation and commands that occur earlier in submission order.
+                gl->glSync.pglMemoryBarrier(c.barrierBits);
```

Comment:
huh?

Reply:
?

Reply:
why do a memory barrier when setting an event?

a Device-only event can only be used within a queue, and OpenGL only cares about future read usages (the other half of a barrier), so it makes sense to only barrier upon a wait

The story would be different here if you supported querying an event on the host, then certain barrier bits at event signal time would make sense.

Reply:
yea right, that's stupid XD my bad

Reply:
not stupid.. good to have it commented out and hanging around so we dont fuckup barriering on GL_CLIENT_MAP__BARRIER_BIT etc.